torch/_inductor/codegen/cpp_mlp_template.py

```diff
@@ -0,0 +1,650 @@
+# mypy: allow-untyped-defs
+import contextlib
+import logging
+from typing import Callable, List, Optional, Set, Union
+from unittest.mock import patch
+
+import torch
+import torch.utils
+
+from ..._dynamo.utils import counters
+from .. import config, ir, lowering as L
+from ..kernel.mm_common import mm_args
+from ..select_algorithm import DataProcessorTemplateWrapper
+from ..utils import parallel_num_threads
+from ..virtualized import V
+from .cpp import get_export_declaration
+from .cpp_gemm_template import CppPackedGemmTemplate, get_padded_n
+from .cpp_micro_gemm import CppMicroGemmAMX, create_micro_gemm, LayoutType
+from .cpp_template_kernel import CppTemplateKernel
+from .cpp_utils import (
+    DTYPE_TO_CPP,
+    GemmBlocking,
+    get_gemm_template_output_and_compute_dtype,
+)
+
+
+log = logging.getLogger(__name__)
+
+
+# TODO<Leslie>: We should merge this with GEMM Template to save redundant code
```

Comment:
Is it better to do some refactoring before this PR?

Reply:
Thanks for the comment. Move the common template code into the standalone sub-template. So, it can be reused by different template.